pkg/workers/sharings/share_data.go

```diff
@@ -98,12 +99,16 @@ func (opts *SendOptions) fillDetailsAndOpenFile(fs vfs.VFS, fileDoc *vfs.FileDoc
 	fileOpts.md5 = base64.StdEncoding.EncodeToString(fileDoc.MD5Sum)
 
 	// Send references for permissions
-	// TODO: only send the reference linked to the actual permission
-	b, err := json.Marshal(fileDoc.ReferencedBy)
-	if err != nil {
-		return nil
+	var refs string
+	if opts.Selector == consts.SelectorReferencedBy {
+		sharedRefs := opts.getSharedReferences()
+		b, err := json.Marshal(sharedRefs)
+		if err != nil {
+			return err
+		}
+		refs = string(b[:])
```

Comment:
@nono From what I understood, a byte array cannot be direclty converted to string, because it is actually a slice. So you need to first convert the byte array to slice using `[:]` , then convert to string.